test/kennel/parts_writer_test.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+require_relative "../test_helper"
+require "find"
+
+SingleCov.covered!
+
+describe Kennel::PartsWriter do
+  def write(file, content)
+    folder = File.dirname(file)
+    FileUtils.mkdir_p folder unless File.exist?(folder)
+    File.write file, content
+  end
+
+  let(:project_filter) { nil }
+  let(:tracking_id_filter) { nil }
+
+  let(:filter) do
+    filter = "some filter".dup
+    filter.stubs(:project_filter).returns(project_filter)
+    filter.stubs(:tracking_id_filter).returns(tracking_id_filter)
+    filter
+  end
+
+  capture_all
+
+  def make_project(kennel_id, monitor_kennel_ids)
```

Comment:
prefer to have all lets / all defs / all setups + 'in_temp_dir / capture_all' in a single place and not interleaved

Reply:
What do you mean? All the defs together, all the lets together, etc?

Reply:
something like
```
def foo
def bar

let(:baz)
let(:fop)

in_tempdir
before { stuff }
after { stuff }
```

Reply:
d813cfb4905f515250f9131e9cc047d3ddac0081